### PR TITLE
EOS-18598: Shutdown S3 services, during reset phase

### PR DESCRIPTION
**Pull request by knrajnambiar76**
Shutdown s3 services during reset phase

Signed-off-by: Rajesh <rajesh.nambiar@seagate.com>

### CODE REVIEW

File: scripts/provisioning/s3_setup [hunk 1]
```diff
@@ -41,7 +41,7 @@ test                     run s3 sanity tests
 
 cleanup                  cleanup S3 account created for backgrounddelete service
 
-reset                    delete all logs related to S3
+reset                    shutdown various s3 services and delete all logs related to S3
```
Comment (nileshgovande) [inline scripts/provisioning/s3_setup:44]: Stop

File: scripts/provisioning/resetcmd.py [hunk 2]
```diff
@@ -33,8 +35,56 @@ def __init__(self, config: str):
...
+    cmd = '/bin/systemctl stop %s.service' % s3service
+    proc = subprocess.Popen(cmd, shell=True,stdout=subprocess.PIPE)
+    proc.communicate()
+    if proc.returncode==0:
+      print('stopped service ', s3service)
+    else:
+      print('Failed to stop service ', s3service)
+    return proc.returncode
+
+  def stop_s3services(self):
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:72]: Array + loop

File: scripts/provisioning/resetcmd.py [hunk 3]
```diff
@@ -33,8 +35,56 @@ def __init__(self, config: str):
...
+    stdout_list = proc.communicate()[0].split('\n')
+    for line in stdout_list:
+        if 'Active:' in line:
+            if '(running)' in line:
+                return True
+    return False
+
+  def stop_active_service(self):
+    """Stop service and return returncode"""
+    cmd = '/bin/systemctl stop %s.service' % self.service
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:52]: CMDSTATUS,  CMDSTOP constants

File: scripts/provisioning/resetcmd.py [hunk 4]
```diff
@@ -33,8 +35,56 @@ def __init__(self, config: str):
     except Exception as e:
       raise e
 
+  def is_service_active(self, s3service):
+    """Return True if service is running"""
+    self.service = s3service
+    cmd = '/bin/systemctl status %s.service' % self.service
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:41]: use framework like - /root/workspace/cortx-s3server/scripts/provisioning/testcmd.py

File: scripts/provisioning/resetcmd.py [hunk 5]
```diff
@@ -19,9 +19,12 @@
 #
 
 import sys
-
+from cortx.utils.process import SimpleProcess
 from setupcmd import SetupCmd
 
+services_list = ["haproxy", "s3backgroundconsumer", "s3authserver"]
+all_services_instances_list = ["s3backgroundproducer@*", "s3server@*"]
```
Comment (nileshgovande) [inline scripts/provisioning/resetcmd.py:26]: this service file is not in use yet - so make it  's3backgroundproducer' and move it to 'services_list'.

File: scripts/provisioning/cleanupcmd.py [hunk 6]
```diff
@@ -68,6 +68,15 @@ def process(self):
     """Main processing function."""
     sys.stdout.write(f"Processing {self.name} {self.url}\n")
     self.phase_prereqs_validate(self.name)
+
+    try:
+      sys.stdout.write("checking if ldap service is running or not...\n")
+      if not self.is_service_active("slapd"):
+      self.restart_service("slapd")
```
Comment (nileshgovande) [inline scripts/provisioning/cleanupcmd.py:75]: stop slapd service at the end of cleanup and clear its logfile

File: scripts/provisioning/setupcmd.py [hunk 7]
```diff
@@ -41,6 +42,8 @@ class SetupCmd(object):
   hosts_list = None
   s3_prov_config = "/opt/seagate/cortx/s3/mini-prov/s3_prov_config.yaml"
   _preqs_conf_file = "/opt/seagate/cortx/s3/mini-prov/s3setup_prereqs.json"
+  services_list = ["haproxy","s3backgroundconsumer","slapd","s3authserver"]
+  all_services_instances_list = ["s3backgroundproducer@*", "s3server@*"]
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:46]: Use  's3backgroundproducer' and move to services_list

File: scripts/provisioning/setupcmd.py [hunk 8]
```diff
@@ -145,3 +148,72 @@ def phase_prereqs_validate(self, phase_name: str):
                                 files=_prereqs_confstore.get_config(f'{phase_name}>files'))
     except Exception as e:
       raise S3PROVError(f'ERROR: {phase_name} prereqs validations failed, exception: {e} \n')
+
+  def is_service_active(self, s3service):
+    """Return True if service is running."""
+    self.service = s3service
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:154]: Use validator instead

File: scripts/provisioning/setupcmd.py [hunk 9]
```diff
@@ -145,3 +148,72 @@ def phase_prereqs_validate(self, phase_name: str):
...
+    handler = SimpleProcess(cmd)
+    sys.stdout.write(f"shutting down {self.service}\n")
+    res_op, res_err, res_rc = handler.run()
+    if res_rc != 0:
+      raise Exception(f"{cmd} failed with err: {res_err}, out: {res_op}, ret: {res_rc}")
+    else:
+      return res_rc
+
+  def shutdown_service(self, s3service):
+    """Stop service and return returncode."""
```
Comment (nileshgovande) [inline scripts/provisioning/setupcmd.py:181]: Pass a list - it can have one or more elements

File: scripts/provisioning/s3_setup (hunk 1 above)
Comment (sachitanands) [reply scripts/provisioning/s3_setup]: done

File: scripts/provisioning/resetcmd.py (hunk 2 above)
Comment (sachitanands) [reply scripts/provisioning/resetcmd.py]: done

File: scripts/provisioning/resetcmd.py (hunk 4 above)
Comment (sachitanands) [reply scripts/provisioning/resetcmd.py]: done

File: scripts/provisioning/resetcmd.py (hunk 5 above)
Comment (sachitanands) [reply scripts/provisioning/resetcmd.py]: added to list

File: scripts/provisioning/cleanupcmd.py (hunk 6 above)
Comment (sachitanands) [reply scripts/provisioning/cleanupcmd.py]: done

File: scripts/provisioning/setupcmd.py (hunk 7 above)
Comment (sachitanands) [reply scripts/provisioning/setupcmd.py]: done

File: scripts/provisioning/setupcmd.py (hunk 8 above)
Comment (sachitanands) [reply scripts/provisioning/setupcmd.py]: done

File: scripts/provisioning/setupcmd.py (hunk 9 above)
Comment (sachitanands) [reply scripts/provisioning/setupcmd.py]: done

**Comment by sachitanands** [comment]
addressed review comments and now testing is pending for these changes

File: scripts/provisioning/resetcmd.py [hunk 10]
```diff
@@ -28,6 +28,8 @@
 from s3backgrounddelete.cortx_s3_constants import MESSAGE_BUS
 from setupcmd import SetupCmd
 
+services_list = ["haproxy", "s3backgroundconsumer", "s3backgroundproducer", "slapd", "s3authserver", "s3server@*"]
```
Comment (knrajnambiar76) [inline scripts/provisioning/resetcmd.py:31]: you can have s3server@* after s3backgroundproducer and also slapd after s3authserver -- Basically shutting producer before consumer

Comment (sachitanands) [reply scripts/provisioning/resetcmd.py]: done

**Review by vimalraghwani20 — APPROVED** [review]

**Review by nileshgovande — APPROVED** [review]

**Pull request merged**